lib/dor/workflow/client/status.rb

```diff
@@ -44,23 +44,26 @@ def initialize(druid:, version:, lifecycle_routes:)
         # @return [Hash{Symbol => Object}] including :status_code and :status_time
         def info
           @info ||= begin
-            # if we have an accessioned milestone, this is the last possible step and should be the status regardless of time stamp
             accessioned_milestones = current_milestones.select { |m| m[:milestone] == 'accessioned' }
-            return { status_code: STEPS['accessioned'], status_time: accessioned_milestones.last[:at].utc.xmlschema } unless accessioned_milestones.empty?
-
-            status_code = 0
-            status_time = nil
-            # for each milestone in the current version, see if it comes at the same time or after the current 'last' step, if so, make it the last and record the date/time
-            current_milestones.each do |m|
-              m_name = m[:milestone]
-              m_time = m[:at].utc.xmlschema
-              next unless STEPS.key?(m_name) && (!status_time || m_time >= status_time)
-
-              status_code = STEPS[m_name]
-              status_time = m_time
+            if accessioned_milestones.size >= 1
+              # if we have an accessioned milestone, this is the last possible step and should be the status regardless of timestamp
+              { status_code: STEPS['accessioned'], status_time: accessioned_milestones.last[:at].utc.xmlschema }
```

Comment:
I'm not sure what specific cop(s) wanted these changes, but I might prefer the guard clause approach here. How about:

```ruby
return { status_code: STEPS['accessioned'], status_time: accessioned_milestones.last[:at].utc.xmlschema } if accessioned_milestones.any?
```

Reply:
@mjgiarlo rubocop squawked at a return statement inside a begin block.   I was trying to avoid that and a bunch of other things.  Maybe there's a better way tho -- this code has a high cognitive load.